Packs/qualys/Integrations/QualysEventCollector/QualysEventCollector.py

```diff
@@ -0,0 +1,638 @@
+        max_fetch: max number of events to return
+        next_page: pagination marking
+    Returns:
+        Host list detections events, Next run datetime
+    """
+    demisto.debug(f'Starting to fetch host list events: last_time={last_time}, next_page={next_page}')
+
+    host_list_detections = client.get_host_list_detection(since_datetime=last_time, max_fetch=max_fetch, next_page=next_page)
+    host_list_events, next_url = handle_host_list_detection_result(host_list_detections) or []
+    newest_event_time = host_list_events[0].get('LAST_VM_SCANNED_DATE') if host_list_events else last_time
+
+    new_next_page = get_next_page_from_url(next_url, 'id_min')
+
+    if newest_event_time == last_time:
+        edited_host_detections = []
+    else:
+        edited_host_detections = get_detections_from_hosts(host_list_events)
+        demisto.debug(f'Parsed detections from hosts, got {len(edited_host_detections)=} events.')
+
+        edited_host_detections = remove_events_before_last_scan(edited_host_detections, last_time)
+
+        add_fields_to_events(edited_host_detections, ['DETECTION', 'FIRST_FOUND_DATETIME'], 'host_list_detection')
+
+    next_run_dict = {
+        HOST_LAST_FETCH: datetime.now().strftime(DATE_FORMAT) if not new_next_page else None,
+        HOST_DETECTIONS_NEWEST_EVENT_DATETIME: newest_event_time,
+        HOST_DETECTIONS_NEXT_PAGE: new_next_page,
+        HOST_DETECTIONS_SINCE_DATETIME_PREV_RUN: last_time,
+    }
+    demisto.debug(f'Done to fetch host list events: {next_run_dict=}, sending {len(edited_host_detections)} events.')
+
+    return edited_host_detections, next_run_dict
+
+
+def fetch_events(client, last_run, first_fetch_time, fetch_function, newest_event_field, next_page_field,
+                 previous_run_time_field, max_fetch: int | None = 0):
+    """ Fetches activity logs and host list detections
+    Args:
+        client: command client
+        last_run: last fetch time
+        first_fetch_time: when start to fetch from
+        fetch_function: function that gets the events
+        max_fetch: max number of items to return (0 to return all)
+        newest_event_field
+        next_page_field
+        previous_run_time_field
+    Return:
+        next_last_run: where to fetch from next time
+        event: events to push to xsiam
+    """
+    demisto.debug(f'Starting fetch for {fetch_function.__name__}, last run: {last_run}')
+    newest_event_time = last_run.get(newest_event_field) if last_run else None
+    next_page = last_run.get(next_page_field)
+    previous_time_field = last_run.get(previous_run_time_field)
+
+    if not newest_event_time:
+        newest_event_time = first_fetch_time
+
+    time_to_fetch = newest_event_time if not next_page else previous_time_field
+
+    events, new_next_run = fetch_function(client, time_to_fetch, max_fetch, next_page)
+
+    updated_next_run = {previous_run_time_field: new_next_run.get(previous_run_time_field)}
```

Comment:
isn't it equivalent to:
```suggestion
    updated_next_run = {previous_run_time_field: time_to_fetch}
```

Reply:
also, why not just use `new_next_run` instead of `updated_next_run`? it's a bit confusing to follow